src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -943,7 +943,20 @@ public void MarkSymbolEndAnalysisComplete(ISymbol symbol, DiagnosticAnalyzer ana
                 {
                     ExecuteBlockActionsCore<CodeBlockStartAnalyzerAction<TLanguageKindEnum>, CodeBlockAnalyzerAction, SyntaxNodeAnalyzerAction<TLanguageKindEnum>, SyntaxNodeAnalyzerStateData, SyntaxNode, TLanguageKindEnum>(
                         codeBlockStartActions, codeBlockActions, codeBlockEndActions, analyzer,
-                        declaredNode, declaredSymbol, executableCodeBlocks, (codeBlocks) => codeBlocks.SelectMany(cb => cb.DescendantNodesAndSelf()),
+                        declaredNode, declaredSymbol, executableCodeBlocks, (codeBlocks) => codeBlocks.SelectMany(
+                            cb =>
+                            {
+                                var filter = semanticModel.GetSyntaxNodesToAnalyzeFilter(cb, declaredSymbol);
```

Comment:
Is the below existing invocation to this method redundant now? https://github.com/dotnet/roslyn/blob/507c81a5146ba714f827b730cac2339605e22c12/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L2695-L2697

Reply:
> Is the below existing invocation to this method redundant now? [https://github.com/dotnet/roslyn/blob/507c81a5146ba714f827b730cac2339605e22c12/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L2695-L2697](https://github.com/dotnet/roslyn/blob/507c81a5146ba714f827b730cac2339605e22c12/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L2695-L2697)

No, that place iterates through nodes itself

---
In reply to: [461205407](https://github.com/dotnet/roslyn/pull/46308#discussion_r461205407) [](ancestors = 461205407)